pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -3766,13 +3773,26 @@ func (ss *SemaphoreStatus) LockWaiting(holderKey, lockKey string, currentHolders
 	return true
 }
 
+func GetHolderKeyVersion() HoldingNameVersion {
+	switch os.Getenv("HOLDER_KEY_VERSION") {
+	case "v2":
+		return HoldingNameV2
+	default:
+		return HoldingNameV1
+	}
+}
+func GetHoldingNameV2OrV1(holderKey string) string {
+	switch GetHolderKeyVersion() {
+	case HoldingNameV2:
+		return GetHoldingNameV2(holderKey)
+	default:
+		return GetHoldingNameV1(holderKey)
+	}
+}
+
```

Comment:
These could move into sync_manager. That way they do not need to be  exported.

Reply:
Yes, do this.

Fewer exports is almost always the right answer.